### PR TITLE
Bug 1107674 - Gaia UI Test (Privileged) for Settings API Access

### PR DESCRIPTION
**Pull request by qdot**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[qdot Kyle Machulis (qdot)](https://github.com/qdot) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=9d29c612a861)

File: dev_apps/uitest-privileged/js/API/settings.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  addStatus: function (aStatus) {
+    this.statusPlaceholder.innerHTML = this.statusPlaceholder.innerHTML + "<br\>" + aStatus;
+  },
+  onError: function(aError) {
+    console.error('ERROR ' + aError.name || aError);
+    this.errorContainer.classList.remove('hidden');
+    this.errorPlaceholder.textContent = aError.name;
+  },
+  runTest: function() {
+    if (navigator.mozSettings)
```
Comment (lissyx) [inline dev_apps/uitest-privileged/js/API/settings.js]: nit: maybe we should bail early and not do the rest of the logic in the body. Like:
if (!navigator.mozSettings) {
  this.onError(...);
  return;
}

Just to make it easier to read.

File: dev_apps/uitest-privileged/js/API/settings.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        req2.onerror = (() => {
+          this.onError("COULD NOT SET WALLPAPER");
+        });
+      });
+      req.onerror = (() => {
+        this.onError("COULD NOT RETRIEVE WALLPAPER.")
+      });
+    }
+    else {
+      this.onError("SETTINGS API UNAVAILABLE");
```
Comment (lissyx) [inline dev_apps/uitest-privileged/js/API/settings.js]: If I remove the permission from the manifest, then I see:
01-13 17:45:29.719  4056  4056 E UI tests - Privileged App: Content JS ERROR: ERROR undefined 
01-13 17:45:29.719  4056  4056 E UI tests - Privileged App:     at settingsTest.onError (app://uitest-privileged.gaiamobile.org/js/API/settings.js:13:4)
01-13 17:45:29.719  4056  4056 E UI tests - Privileged App:     at settingsTest.runTest (app://uitest-privileged.gaiamobile.org/js/API/settings.js:38:6)
01-13 17:45:29.719  4056  4056 E UI tests - Privileged App:     at onload (app://uitest-privileged.gaiamobile.org/js/API/settings.js:46:2)

and the UI is not updated accordingly

**Comment by try-server-hook** [comment]
[qdot Kyle Machulis (qdot)](https://github.com/qdot) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=48eb60694609)

**Comment by try-server-hook** [comment]
[qdot Kyle Machulis (qdot)](https://github.com/qdot) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f4ce11b187bb)

**Comment by qdot** [comment]
Tests green, r+'d by @lissyx, merging.

**Pull request merged**